index.js

```diff
@@ -196,14 +209,21 @@ function colorDelta(img1, img2, k, m, yOnly) {
         b2 = blend(b2, a2);
     }
 
-    const y = rgb2y(r1, g1, b1) - rgb2y(r2, g2, b2);
+    const y1 = rgb2y(r1, g1, b1);
+    const y2 = rgb2y(r2, g2, b2);
+    const y = y1 - y2;
+
+    if (y1 > y2) origin = 2;
 
     if (yOnly) return y; // brightness difference only
 
     const i = rgb2i(r1, g1, b1) - rgb2i(r2, g2, b2);
     const q = rgb2q(r1, g1, b1) - rgb2q(r2, g2, b2);
 
-    return 0.5053 * y * y + 0.299 * i * i + 0.1957 * q * q;
+    return {
+      origin,
+      delta: 0.5053 * y * y + 0.299 * i * i + 0.1957 * q * q
+    };
```

Comment:
I haven't measured this, but this change likely makes performance much worse. Instead of returning a number (something easily inlined/optimized by V8), we now allocate a new complex object for every pixel comparison (which occurs millions of times). We'll have to refactor this so that there are no new allocations. One way to do that would be to return a signed delta — since the squared distance can only be positive, we can encode the origin in the result sign.

Reply:
oh that's a great idea 👍